source/state_representation/include/state_representation/space/cartesian/CartesianState.hpp

```diff
@@ -642,6 +642,7 @@ inline state_representation::CartesianStateVariable string_to_cartesian_state_va
   } else {
     throw exceptions::InvalidStateVariableException("Invalid Cartesian state variable: " + variable);
   }
+  __builtin_unreachable();
```

Comment:
this (and following instances) are to let the compiler know we never get there (because in the else it throws and doesn't return we get a warning).

Reply:
Could also just remove the else and have the throw be the last line of the function no? same effect though so i don't mind

Reply:
that would work, but seeing as we are probably using it for the `joint_state_variable_to_string` I would probably keep this for all functions. I personally actually kind of like that very explicit line, but no strong opinion on that, it's a small detail